Prevent chests demanding items unless there is space

dave-mills opened the pull request:
Hi! Firstly, this is a great mod - I like that it keeps a pretty vanilla feel while adding great auto-sorting functions. 

I was wondering - are you open to PRs for this? I have a couple of things I've been tinkering with while trying this mod out on a couple of worlds. This PR is for one of them (they're pretty separate features so I thought I should separate them, in case you want to include one but not the other.) 

This adds a check to see if any given item can fit into the amazing chest's inventory. If it cannot fit, it is rejected, allowing the item to flow to another chest in the hopper chain (e.g. an overflow chest, or a 2nd amazing chest with the same item filter). 

I saw issue #3 related to multiple chests with the same filters, where you were thinking of distributing items evenly. I guess this change is incompatible with that idea, so feel free to reject this PR if it doesn't fit with how you want to develop the mod :) 

Also - I'm very new to Java, so let me know if I've done anything silly here! Would love to learn more about Minecraft modding.

dave-mills committed:
prevent chests demanding items unless there is space

pcal43 commented:
hey sorry for the delayed response, have been super busy IRL.  thanks for sending this.  

looks good on first glance.    i'll kick the tires and try to merge next week.

dave-mills commented:
Thanks! 

One thing I have noticed since posting is that when there are a bunch of the chests, it can get laggy when items are going through the hopper chains - probably because of the fullness checking... not sure what to do about that - I haven't figured out the caching stuff, but maybe it could hook into that so it isn't constantly checking inventories? If you have any thoughts on that, let me know - happy to do some tinkering.

dave-mills committed:
fix crash when block is above chest:
- it appears a chest inventory is null when it cannot be opened due to a full block being placed above it

dave-mills committed:
tweak default hasSpaceFor response
